tests/functional/z_cluster/test_performance_profile_validation.py

```diff
@@ -142,10 +192,15 @@ def test_change_cluster_resource_profile(self, perf_profile):
 
             verify_storage_cluster()
 
-            # Verify that storage cluster is updated with new profile
+            # Sleep for 120 seconds for the profile changes to reflect
+            time.sleep(120)
```

Comment:
Here I would use `TimeoutSampler` to wait for the profile to change to `reflect`.